java/flight/flight-core/pom.xml

```diff
@@ -323,7 +314,6 @@
         <dependency>
           <groupId>io.netty</groupId>
           <artifactId>netty-transport-native-unix-common</artifactId>
-          <version>${dep.netty.version}</version>
```

Comment:
From some searching, it seems dependencyManagement may not get inherited by profiles, so we'll need to explicitly reference the version here still.

Reply:
What will be the impact if we only use `io.netty:netty-transport-native-unix-common:4.1.78.Final` instead of `io.netty:netty-transport-native-unix-common:osx-x86_64:4.1.78.Final`? Is there some reason to why we use that? 

I am building and testing code without -osx / -linux and it finished without problems.

Reply:
That said our tests might not actually hit any of the platform-specific paths

Reply:
> That said our tests might not actually hit any of the platform-specific paths

Does it mean that we still need the platform-specific JNI library for actual usages? Or, do we ever need no the JNI libraries for both test and actual usages?

Reply:
We currently do not need the Netty JNI libraries for tests, because we aren't properly testing those paths (ARROW-17024). For actual use, they are only required if the application is using domain sockets, or wants to manually configure Netty to use epoll/kqueue. The application can always explicitly add the necessary dependencies.

Reply:
I see. I will remove [this code](https://github.com/apache/arrow/blob/master/ci/scripts/java_build.sh#L62-L75) to download the Netty JNI binaries from the build script for s390x.

Reply:
Sorry for my bad. These binaries may be still required.